docs(readme): add instructions for adding validator information

onmax opened the pull request.

onmax committed:
docs(readme): add instructions for adding validator information

Eligioo commented on README.md:
```diff
@@ -12,6 +12,19 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+In order to do so, follow these steps:
+
+1. [Fork this repository](https://github.com/nimiq/validators-api/fork).
+2. Create a new file in the [validators](./public/validators/) folder. You can use `.example.json` as a reference.
```
I had a look at the example.json and not all fields were quite clear to me:
- icon: Should this icon be added to the repo too? Or via a publicly reachable domain?
- fee: How is this field interpreted? Can it be int or decimals? Does a value of 1 mean 1% fee?
- isPool: Why is the default `null`? I would expect a boolean here.

onmax replied:
Ok, I can change this file to:

```
{
     "icon": "Optional. A base64 encoded SVG icon. It should start with `data:image/svg+xml;base64...`. If you are not sure how to get this value, you can leave this field empty and add the svg in the PR. One of the maintainers will convert it for you :)",
     "fee": 0.1,
     "isPool": false
}
```

I want to keep this field JSON-compatible, so I will move the comments also to this readme

Eligioo replied:
Alright 👍

Eligioo commented on README.md:
```diff
@@ -12,6 +12,19 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+3. Submit a PR to this repo. Someone in the team will review it.
+
+> [!INFO]
+> By submitting this information, various Nimiq applications, such as the Nimiq Wallet, will show users the fields you have submitted and the Validator Score associated with your validator in the future. You can read more about the Validator Score in the [Developer's Center PR](https://github.com/nimiq/developer-center/pull/21).
```
Linking to a PR isn't really helpful? Can't we link to the developer center directly?

onmax replied:
It is not merged yet. People reading these instructions are quite familiar already with GitHub so I thought it wouldn't be a problem :/

Eligioo replied:
You are looking at raw markup and git changes here. Not exactly pretty but just my 2 cents.

sisou replied:
The preview link in the PR description doesn't work.

Eligioo commented on README.md:
```diff
@@ -12,6 +12,40 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+2. Create a new file in the [validators](./public/validators/) folder. You can use `.example.json` as a reference.
+
+<details>
+  DCFER32W1Q<tab>	YTRFEDWSQAAAAAAAAAAAAAAAAAAAAAA
```
A cat walked over your keyboard? 🐱

Eligioo commented on README.md:
```diff
@@ -12,6 +12,40 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+When you add your validator information, you'll need to include certain keys in your JSON file. Below is an explanation of each key and its possible values:
+
+- `address` (required): The address of the validator in the format `NQXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX`.
+- `fee` (Required): A number between -1 and 1 representing the fee percentage charged by your validator. For example, 0.05 represents a 5% fee.
```
A number between 0 and 1? I suppose -1 when there is no fee?

Eligioo commented on README.md:
```diff
@@ -12,6 +12,40 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+- `fee` (Required): A number between -1 and 1 representing the fee percentage charged by your validator. For example, 0.05 represents a 5% fee.
+- `payoutType` (required): The method used to pay out validator rewards. Possible values are
+  - `restake`: Rewards are automatically restaked.
+  - `direct`: Rewards are paid directly to the validator.
```
```suggestion
  - `direct`: Rewards are paid directly to the staker's wallet and not automatically restaked.
```

Eligioo commented on README.md:
```diff
@@ -12,6 +12,40 @@ Details of validators in the Nimiq Blockchain and their scores, calculated using
+- `name` (optional): The name of the validator. If not specified, the address of the validator is used by default.
+- `description` (optional): A short description of your validator.
+- `website` (optional): The URL of your validator's website.
+- `icon` (optional): A base63 encoded SVG icon representing your validator. It is preferable if it has Nimiq's hexagon shape.
```
```suggestion
- `icon` (optional): A base64 encoded SVG icon representing your validator. The logo being in Nimiq's hexagon shape is preferred.
```

sisou replied:
Representing your **pool**.

Eligioo commented on public/validators/.example.json:
```diff
@@ -1,9 +1,9 @@
   "name": "Optional. The name of the validator, by default it is the address",
   "description": "Optional. A description of the validator",
-  "website": "Optional. The website of the validator",
+  "website": "Optional. The website of the validator. By default identicon is used. Preferable if it has an Nimiq's hexagon-shape",
```
The last bit of this text belongs to the `icon` property of the JSON?

onmax committed:
replaced `isPool` for `payoutType` enum

onmax committed:
replaced `isPool` for `payoutType` enum

onmax committed:
replaced `isPool` for `payoutType` enum

onmax committed:
replaced `isPool` for `payoutType` enum

Albermonte committed:
typos

Albermonte committed:
default is remote, fixed comment

Albermonte committed:
Updating deps

Albermonte committed:
Updating deps

Albermonte committed:
Updating deps

Albermonte committed:
feat(fetch): store missing epoch even if they dont exist
this is happening because on one testnet restart the genesis was the same as before but the data was not keep, it was coming from a snapshot

Albermonte committed:
feat(fetch): store missing epoch even if they dont exist
this is happening because on one testnet restart the genesis was the same as before but the data was not keep, it was coming from a snapshot

Albermonte committed:
feat(fetch): store missing epoch even if they dont exist
this is happening because on one testnet restart the genesis was the same as before but the data was not keep, it was coming from a snapshot

Albermonte committed:
small performance improvement

Albermonte committed:
fix(validators): use icon from validators list

Albermonte committed:
Identicon is used by default if no icon provided

Albermonte committed:
Display reason for the score in validator page

Albermonte committed:
Display reason for the score in validator page

Albermonte committed:
Display reason for the score in validator page

Albermonte committed:
Added size to reason

Albermonte committed:
fix(fetcher): use correct range
First election block is genesis block
Epoch 0 is the genesis block
Batches start at 1
Improved parallelization

Albermonte committed:
fix(fetcher): use correct range
First election block is genesis block
Epoch 0 is the genesis block
Batches start at 1
Improved parallelization

Albermonte committed:
fix(findMissingEpochs): fetch previously latest epoch not marked as missing epoch

Latest epoch is stored with everything as -1, that epoch is also missing and needs to be re-fetched once that's not the latest epoch

Albermonte committed:
style(logs): minor improvements

Albermonte committed:
style(logs): minor improvements

Albermonte committed:
style(logs): minor improvements

Albermonte committed:
updating deps

Albermonte committed:
updating deps

Albermonte committed:
updating deps

Albermonte committed:
feat(release-it): initial setup

Albermonte committed:
feat(release-it): initial setup

Albermonte committed:
feat(release-it): initial setup

Albermonte committed:
fix(release-it): dont publish to npm

Albermonte committed:
added dotenv to release

Albermonte committed:
added dotenv to release

Albermonte committed:
added dotenv to release

Albermonte committed:
feat(gh-actions): running fetch missing epochs every 12 hours

Albermonte committed:
updating dependencies

Albermonte committed:
updating dependencies

Albermonte committed:
updating dependencies

Albermonte committed:
updating deps in package

Albermonte committed:
updating deps in package

Albermonte committed:
feat(api): added endpoints to get a specific epochs or know the missing ones

Albermonte committed:
disable fetchMissing for github actions

Albermonte committed:
feat: fetch missing epochs using github actions

Albermonte committed:
feat(Identicon): using placeholder if no validator has no icon

Albermonte committed:
feat(import-validators): created endpoint to import validators

Albermonte committed:
feat: import validators after fetching missing epochs

Albermonte committed:
fix(release-it): skipping github check
it's giving not a collaborator issue even when I'm an admin

onmax committed:
chore: pnpm lock

onmax committed:
lint

onmax committed:
lint

onmax committed:
chore: updated dependecies

onmax committed:
chore: updated dependecies

onmax committed:
Merge branch 'main' of https://github.com/nimiq/validators-api into onmax/validator-instructions
